src/queue.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+export default class Queue {
+  constructor(initialValues=[]) {
+    this.elements = initialValues
+    this.first = initialValues[0]
+    this.last = initialValues.length
+  }
+    enqueue(value){
+      this.elements[this.last] = value
+      this.last++
+    }
+
+    dequeue(){
+      if(this.elements.length === 0 ){
+        return null
+      }
+      let result = this.elements.slice(0, 1)
+      this.elements.splice(0,1)
+      return result
+    }
```

Comment:
Cleaner implementation:

```javascript
if (this.elements.length) {
  return this.elements.splice(0, 1)
}

return null
```

Alternatively, you can use the built in `Array.shift()` [function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift).